test/lit/passes/gufa-refs.wast

```diff
@@ -5387,7 +5387,7 @@
     (drop
       (block $B (result (ref $B))
         (drop
-          (br_on_cast $B $B
+          (br_on_cast $B anyref (ref $B)
```

Comment:
why not `anyref => (ref $A)` ?

Reply:
For a lot of these tests, it was simpler to just blindly type "anyref" as the input type rather than put thought into what more specific input type would make sense. The input type is thrown away during parsing, so it doesn't affect the IR. If you'd like, I can take another pass and replace "anyref" with more precise types.

Another complication is that the cast type is required to be a subtype of the input type (although so far I haven't implemented this yet in the PR), so I can't always use the precise type of ref operand, either.

Reply:
Sounds good, I was just checking I wasn't missing something.

Reply:
I started implementing this subtyping constraint, but it turns out to be unfortunately complicated, so I filed https://github.com/WebAssembly/gc/issues/381 to try to get the constraint dropped instead.